File: teamB/compiler/bin/javascript_sql_compiler/test_files_js/master_unit_test.js

const testCases = 5
let passed = 0;

const studentFilePath = process.argv[2];
const { addition, subtraction, multiplication, division, power } = require(`./${studentFilePath}`);

function runTest() {
    const addResult = addition(2, 3);
    if (addResult === 5) {
        passed += 1;
        console.log(`Addition test passed: ${addResult} === 5`);
    } else {
        console.log(`Addition test failed: ${addResult} !== 5`);
    }

    const subtractResult = subtraction(3, 1);
    if (subtractResult === 2) {
        passed += 1;
        console.log(`Subtraction test passed: ${subtractResult} === 2`);
    } else {
        console.log(`Subtraction test failed: ${subtractResult} !== 2`);
    }

    const multiplyResult = multiplication(3, 4);
    if (multiplyResult === 12) {
        passed += 1;
        console.log(`Multiplication test passed: ${multiplyResult} === 12`);
    } else {
        console.log(`Multiplication test failed: ${multiplyResult} !== 12`);
    }

    const divideResult = division(10, 2);
    if (divideResult === 5) {
        passed += 1;
        console.log(`Division test passed: ${divideResult} === 5`);
    } else {
        console.log(`Division test failed: ${divideResult} !== 5`);
    }

    const powerResult = power(2, 2);
    if (powerResult === 4) {
        passed += 1;
        console.log(`Power test passed: ${powerResult} === 4`);
    } else {
        console.log(`Power test failed: ${powerResult} !== 4`);
    }

    console.log(`${studentFilePath}: ${passed} out of ${testCases} test cases passed`);
    console.log(`-------------------------------------------------------------------`);
}

runTest();

File: teamB/compiler/bin/javascript_sql_compiler/uploads_python/master_unit_test.py
import sys 
import importlib.util

studentFilename = sys.argv[1]

filename = "uploads_python/" + studentFilename

spec = importlib.util.spec_from_file_location("addition", filename)
module = importlib.util.module_from_spec(spec)
spec.loader.exec_module(module)

addition = getattr(module, "addition")
subtraction = getattr(module, "subtraction")
multiplication = getattr(module, "multiplication")
division = getattr(module, "division")
power = getattr(module, "power")

def runTests():
    testCases = 5
    passed = 0

    addResult = addition(2, 3)
    if addResult == 5:
        print(f"Addition test passed: {addResult} == 5")
        passed += 1
    else:
        print(f"Addition test failed: {addResult} != 5")

    subtractionResult = subtraction(5, 3)
    if subtractionResult == 2:
        print(f"Subtraction test passed: {subtractionResult} == 2")
        passed += 1
    else:
        print(f"Subtraction test failed: {subtractionResult} != 2")

    multiplicationResult = multiplication(2, 3)
    if multiplicationResult == 6:
        print(f"Multiplication test passed: {multiplicationResult} == 6")
        passed += 1
    else:
        print(f"Multiplication test failed: {multiplicationResult} != 6")

    divisionResult = division(6, 3)
    if divisionResult == 2:
        print(f"Division test passed: {divisionResult} == 2")
        passed += 1
    else:
        print(f"Division test failed: {divisionResult} != 2")

    powerResult = power(2, 3)
    if powerResult == 8:
        print(f"Power test passed: {powerResult} == 8")
        passed += 1
    else:
        print(f"Power test failed: {powerResult} != 8")

    print(f'{studentFilename}: {passed} out of {testCases} test cases passed')
    print('-------------------------------------------------------------------')

runTests()